### PR TITLE
Mm 43612

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
Cloud instances with a file limit (Starter and Professional subscriptions) show UI notifying that the limit has been hit and that archived files are unavailable until upgrade. Note that because this requires backend changes, but the ticket for the backend is a separate ticket, I made a separate draft branch based off this one that will run the spinwick tied to the backend changes. It will   be posted here:

Spinwick: https://mattermost-webapp-pr-10903.test.mattermost.cloud
Username | Password
`user-mattermost-webapp-pr-10903@example.mattermost.com` | `Cws@10102020`

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43612

#### Related Pull Requests
- Has server changes: https://github.com/mattermost/mattermost-server/pull/20703/files

#### Screenshots

Full size archived display:

regular-display

Compact size archived and unarchived display:
compact-archived

#### Release Note
```release-note
Cloud instances with a file limit (Starter and Professional subscriptions) show UI notifying that the limit has been hit and that archived files are unavailable until upgrade.
```

### CODE REVIEW

**Review by AGMETEOR — APPROVED** [review]
Code LGTM

**Review by nickmisasi — APPROVED** [review]
Non blocking comment. Code looks great.

File: components/root/root.jsx [hunk 1]
```diff
@@ -265,7 +266,7 @@
             landing = desktopAppDownloadLink;
         }
 
-        if (landing && !BrowserStore.hasSeenLandingPage() && !toResetPasswordScreen && !this.props.location.pathname.includes('/landing') && !window.location.hostname?.endsWith('.test.mattermost.com') && !UserAgent.isDesktopApp()) {
+        if (landing && !BrowserStore.hasSeenLandingPage() && !toResetPasswordScreen && !this.props.location.pathname.includes('/landing') && !this.props.location.href?.includes('.test.mattermost.com') && !UserAgent.isDesktopApp()) {
```
[the remark this answers is not in the document]
Comment (neallred) [reply components/root/root.jsx]: true, but part of a merge request and not part of this PR

**Comment by neallred** [comment]
@stevemudie gentle ping for review

**Review by annelieseklein — APPROVED** [review]
Looks great! Thanks!

**Review by stevemudie — APPROVED** [review]
Looks good @neallred!

**Pull request merged**